src/i18ndude/generator.py

```diff
@@ -76,6 +78,11 @@ def emitStartElement(self, name, attrlist, taldict, metaldict, i18ndict,
                 "i18n:name can only occur inside a translation unit",
                 position)
 
+        if varname and NOT_ALLOWED_IN_NAME.search(varname):
+            raise I18NError(
```

Comment:
None of the raises in this method are covered by tests, so I was unable to add a test for this.